akka-projection-samples/src/main/scala/akka/projection/Samples.scala

```diff
@@ -0,0 +1,168 @@
+ */
+object KafkaAtLeastOnceSample {
+  implicit val system = ActorSystem("sample")
+  implicit val ec = system.dispatcher
+
+  val consumerSettings: ConsumerSettings[String, String] = ???
+  val committerSettings = CommitterSettings(system).withMaxBatch(100)
+
+  val src = KafkaConsumer.atLeastOnce(
+    consumerSettings.withGroupId("group1"),
+    Subscriptions.topics("topic1"),
+    committerSettings) { msg =>
+    Future.successful("pretending that this is projected to somewhere")
+  }
+
+  // a pure Kafka projection is super simple, as it only requires a Source
+  val projection = KafkaProjection(src)
+  projection.start()
+  projection.stop()
+}
+
+object KafkaAtLeastOnceSample2 {
+  implicit val system = ActorSystem("sample")
+  implicit val ec = system.dispatcher
+
+  val consumerSettings: ConsumerSettings[String, String] = ???
+  val committerSettings = CommitterSettings(system).withMaxBatch(100)
+
+  val src = Consumer
+    .committableSource(consumerSettings, Subscriptions.topics("topic1"))
+    .mapAsync(1) { msg =>
+      Future.successful(s"pretending that this is projected to somewhere").map(_ => msg.committableOffset)
+    }
+    .via(Committer.flow(committerSettings))
+
+  // a pure Kafka projection is super simple, as it only requires a Source
+  val projection = KafkaProjection(src)
+  projection.start()
+  projection.stop()
+}
+
+/**
+ * A Kafka projection that commits the offset in the topic before delivering to handler (at-most-once)
+ */
+object KafkaAtMostOnceSample {
+
+  implicit val system = ActorSystem("sample")
+  implicit val ec = system.dispatcher
+
+  val consumerSettings: ConsumerSettings[String, String] = ???
+
+  val src =
+    Consumer.atMostOnceSource(consumerSettings.withGroupId("group1"), Subscriptions.topics("topic1")).mapAsync(1) {
+      msg =>
+        println(s"pretending that this is projected to somewhere")
+        Future.successful(Done)
+    }
+
+  // a pure Kafka projection is super simple, as it only requires a Source
+  val projection = KafkaProjection(src)
+  projection.start()
+  projection.stop()
+}
+
+/**
+ * Sample showing a rough API for consuming events from Akka Persistence Cassandra and saving the
+ * projected model and the offset in a JDBC table
+ */
+object AkkaPersistenceCassandraWithTransactionalJDBCSample {
+
+  implicit val system = ActorSystem("sample")
+  implicit val ec = system.dispatcher
+
+  // we need a function that given an optional offset, returns a Source
+  def sourceProvider(offset: Option[Offset]): Source[EventEnvelope, _] = {
+    val queries = PersistenceQuery(system).readJournalFor[CassandraReadJournal](CassandraReadJournal.Identifier)
+    queries.eventsByTag("someTag", offset.getOrElse(Offset.noOffset))
+  }
+
+  val projection =
+    JdbcProjection.transactional(
+      sourceProvider = sourceProvider,
+      offsetStore = new JdbcOffsetStoreAkkaOffset(projectionId = "demo"),
+      offsetExtractor = (ev: EventEnvelope) => ev.offset) { (connec, envelope) =>
```

Comment:
And we need an extractor from envelope to offset so we can extract it to save in the offset store.